qa/restart-upgrade/src/test/java/org/opensearch/neuralsearch/bwc/SemanticSearch.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.neuralsearch.bwc;
+
+import com.carrotsearch.randomizedtesting.RandomizedTest;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Map;
+import java.util.ArrayList;
+
+
+import static org.opensearch.neuralsearch.TestUtils.NODES_BWC_CLUSTER;
+
+import static org.opensearch.neuralsearch.TestUtils.getModelId;
+import static org.opensearch.neuralsearch.TestUtils.TEXT_EMBEDDING_PROCESSOR;
+import org.opensearch.neuralsearch.query.NeuralQueryBuilder;
+
+
+public class SemanticSearch extends AbstractRestartUpgradeRestTestCase{
+
+    private static final String PIPELINE_NAME = "nlp-pipeline";
+    private static String DOC_ID = "0";
+    private static final String TEST_FIELD = "passage_text";
+    private static final String TEXT= "Hello world";
+
+    //Test restart-upgrade Semantic Search
+    //Create Text Embedding Processor, Ingestion Pipeline and add document
+    //Validate process , pipeline and document count in restart-upgrade scenario
+    public void testSemanticSearch() throws Exception{
+        waitForClusterHealthGreen(NODES_BWC_CLUSTER);
+
+        if (isRunningAgainstOldCluster()){
+            String modelId= uploadTextEmbeddingModel();
+            loadModel(modelId);
+            createPipelineProcessor(modelId,PIPELINE_NAME);
+            createIndexWithConfiguration(
+                    testIndex,
+                    Files.readString(Path.of(classLoader.getResource("processor/IndexMappings.json").toURI())),
+                    PIPELINE_NAME
+            );
+            addDocument(testIndex, DOC_ID,TEST_FIELD,TEXT);
+        }else {
+            Map<String,Object> pipeline= getIngestionPipeline(PIPELINE_NAME);
+            assertNotNull(pipeline);
+            String modelId=getModelId(pipeline, TEXT_EMBEDDING_PROCESSOR);
+            validateTestIndex(modelId);
+            deleteIndex(testIndex);
```

Comment:
pipelines and models are global, please make sure they are not conflicting in case of multiple tests. We either need to wipe them after each test of check before creating if that resource exists. For models I think delete is the better option as models are taking memory, so with certain number of tests we'll start having CB exceptions of model load.

Reply:
Ack